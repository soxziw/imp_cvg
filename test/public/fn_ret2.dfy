method Incr(x: int) returns (r: int)
  ensures r == x + 1
{
  var result : int;
  result := x + 1;
  return result;
}

// verified
method Foo(x: int) returns (r: int)
  requires (x < 0)
{
  var y : int;
  y := x;
  assert (y == x);
  y := Incr(y);
  assert (y == x + 1);
  return 0;
}

// not verified
method Bar(x: int) returns (r: int)
  requires (x < 0)
{
  var y : int;
  y := x;
  y := Incr(y);
  assert (y == x);
  assert (y == x + 1);
  return 0;
}