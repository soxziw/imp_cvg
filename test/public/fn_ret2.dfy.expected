Incr: verified
Foo: verified
Bar: not verified
